tests/js/client/server_parameters/test-export-shard-usage-metrics-bytes-per-shard-per-user-cluster.js

```diff
@@ -0,0 +1,1837 @@
+            db._dropDatabase(name);
+          } catch (err) {
+          }
+        });
+      }
+    },
+    
+    testHasMetricsWhenTruncating : function () {
+      [1, 2].forEach((replicationFactor) => {
+        const cn = getUniqueCollectionName();
+
+        let c = db._create(cn, {replicationFactor});
+        try {
+          let shards = c.shards();
+          assertEqual(1, shards.length);
+
+          // truncate an empty collection
+          c.truncate();
+        
+          let parsed = getParsedMetrics(db._name(), cn);
+          assertFalse(parsed.hasOwnProperty("reads"), {parsed});
+          assertFalse(parsed.hasOwnProperty("writes"), {parsed});
+        
+          // truncate a small collection
+          const n = 100;
+          db._query(`FOR i IN 1..${n} INSERT {value: i} INTO ${cn}`);
+        
+          parsed = getParsedMetrics(db._name(), cn);
+          // count 40-50 bytes for each insert
+          assertTrue(parsed.writes[shards[0]] > n * 40 * replicationFactor, {parsed, shards, replicationFactor});
+          assertTrue(parsed.writes[shards[0]] < n * 50 * replicationFactor, {parsed, shards, replicationFactor});
+        
+          c.truncate();
+         
+          parsed = getParsedMetrics(db._name(), cn);
+          // count 10-20 bytes for each remove
+          assertTrue(parsed.writes[shards[0]] > n * 40 * replicationFactor + n * 10 * replicationFactor, {parsed, shards, replicationFactor});
+          assertTrue(parsed.writes[shards[0]] < n * 50 * replicationFactor + n * 20 * replicationFactor, {parsed, shards, replicationFactor});
+        } finally {
+          db._drop(cn);
+        }
+      });
+    },
+    
+    testHasMetricsWhenTruncatingLarge : function () {
+      [1, 2].forEach((replicationFactor) => {
+        const cn = getUniqueCollectionName();
+
+        let c = db._create(cn, {replicationFactor});
+        try {
+          let shards = c.shards();
+          assertEqual(1, shards.length);
+
+          const n = 40000;
+          db._query(`FOR i IN 1..${n} INSERT {value: i} INTO ${cn}`);
+        
+          let parsed = getParsedMetrics(db._name(), cn);
+          // count 40-50 bytes for each insert
+          assertTrue(parsed.writes[shards[0]] > n * 40 * replicationFactor, {parsed, shards, replicationFactor});
+          assertTrue(parsed.writes[shards[0]] < n * 50 * replicationFactor, {parsed, shards, replicationFactor});
+        
+          c.truncate();
+         
+          parsed = getParsedMetrics(db._name(), cn);
+          // truncate will have performed a DeleteRange - metrics should not have changed!
+          assertTrue(parsed.writes[shards[0]] > n * 40 * replicationFactor, {parsed, shards, replicationFactor});
+          assertTrue(parsed.writes[shards[0]] < n * 50 * replicationFactor, {parsed, shards, replicationFactor});
+        } finally {
+          db._drop(cn);
+        }
+      });
+    },
+   
+    testHasMetricsWhenInsertIntoSmartGraph : function () {
```

Comment:
Similar to my other graph related comment. What about the other graph types. Do they require separate tests?